src/Storefront/Page/Address/Listing/AddressListingCriteriaEvent.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Storefront\Page\Address\Listing;
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Event\NestedEvent;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+class AddressListingCriteriaEvent extends NestedEvent
+{
+    /**
+     * @var Criteria
+     */
+    private $criteria;
+
+    /**
+     * @var SalesChannelContext
+     */
+    private $salesChannelContext;
+
+    public function __construct(Criteria $criteria, SalesChannelContext $salesChannelContext)
+    {
+        $this->criteria            = $criteria;
+        $this->salesChannelContext = $salesChannelContext;
+    }
+
+    /**
+     * @return Criteria
+     */
+    public function getCriteria(): Criteria
+    {
+        return $this->criteria;
+    }
+
```

Comment:
a getter for the salesChannelContext is missing

Reply:
@mitelg I have added the getter